### PR TITLE
Enable direct control of LKokkos backends and add the multi-device support to docs

### PR DESCRIPTION
**Pull request by maliasadi**
This PR updates the docs specifying the required Make arguments to build supported backend devices along with adding `CMAKE_ARGS` to `runtime/Makefile` that enables the end user to pass any compiler arguments or override the preset ones in the make commands. This would facilitate passing/defining device specific compiler arguments and a better support for Kokkos GPUs execution kernels.

### CODE REVIEW

File: runtime/Makefile [hunk 1]
```diff
@@ -10,11 +10,12 @@ QIR_STDLIB_INCLUDES_DIR?=$(MK_DIR)/qir-stdlib/target/release/build/include/
 CODE_COVERAGE?=OFF
 BUILD_TYPE?=Release
 ENABLE_WARNINGS?=ON
-ENABLE_OPENMP?=ON
 ENABLE_LIGHTNING_KOKKOS?=OFF
 ENABLE_OPENQASM?=OFF
 LIGHTNING_GIT_TAG_VALUE?=latest_release
 LIGHTNING_KOKKOS_GIT_TAG_VALUE?=latest_release
+CMAKE_ARGS ?= ""
```
Comment (mlxd) [inline runtime/Makefile:17]: Do the spaces matter here, or is that fine?

File: runtime/README.rst [hunk 2]
```diff
@@ -107,15 +107,31 @@ Installation
...
+    ENABLE_LIGHTNING_KOKKOS=ON make runtime
 
 Lightning-Kokkos provides support for other Kokkos backends including OpenMP, HIP and CUDA.
 Please refer to `the installation guideline <https://github.com/PennyLaneAI/pennylane-lightning-kokkos#installation>`_ for the requirements.
+You can further use the ``CMAKE_ARGS`` flag to issue any additional compiler arguments or override the preset ones in the make commands.
+To build the runtime with Lightning-Kokkos and the ``Kokkos::OpenMP`` backend execution space:
+
+.. code-block:: console
+
+    ENABLE_LIGHTNING_KOKKOS=ON CMAKE_ARGS="-DKokkos_ENABLE_OPENMP=ON" make runtime
```
Comment (mlxd) [inline runtime/README.rst:124]: Perfect --- this allows direct control of the LK backend. TY

**Review by mlxd — COMMENTED** [review]
Just one query, and nothing else. Thanks @maliasadi

File: runtime/Makefile (hunk 1 above)
Comment (maliasadi) [reply runtime/Makefile:17]: It worked with spaces. But will remove spaces to make it consistent with other vars. Thanks!

**Review by mlxd — APPROVED** [review]

File: doc/dev/installation.rst [hunk 3]
```diff
@@ -151,6 +151,10 @@ you can build ``qir-stdlib`` and the runtime from the top level directory:
 
   make runtime
 
+The runtime supports multiple backend devices, enabling the execution of quantum
+circuits locally on CPUs and GPUs, and remotely on Amazon Braket NISQ hardware.
+A list of supported backends along with Make arguments for each device
+are available in the `Catalyst Runtime <https://docs.pennylane.ai/projects/catalyst/en/latest/modules/runtime.html>`_ page.
```
Comment (maliasadi) [inline doc/dev/installation.rst:157]:
```suggestion
A list of supported backends, along with Make arguments for each device, is available in the `Catalyst Runtime <https://docs.pennylane.ai/projects/catalyst/en/latest/modules/runtime.html>`_ page.
```

**Pull request merged**